marconi-chain-index/test/Spec/Marconi/ChainIndex/Experimental/Indexers/MintTokenEvent.hs

```diff
@@ -0,0 +1,483 @@
+propRunnerDoesntTrackUnselectedAssetId :: H.Property
+propRunnerDoesntTrackUnselectedAssetId = H.property $ do
+  events <- H.forAll genTimedEvents
+  let allAssetIds = getAssetIdsFromTimedEvents $ mapMaybe sequence events
+
+  -- We take a subset of the AssetIds to track them. If the subset is empty, we track all the
+  -- assetIds.
+  let notAllAddressesPredicate xs = all ($ xs) [not . null, (/= allAssetIds)]
+  trackedAssetIds <- H.forAll $ H.Gen.filter notAllAddressesPredicate $ H.Gen.subsequence allAssetIds
+  let untrackedAssetIds = allAssetIds \\ trackedAssetIds
+
+  sqlIndexer <- indexWithRunner trackedAssetIds events
+
+  forM_ untrackedAssetIds $ \untrackedAssetId -> do
+    let query =
+          case untrackedAssetId of
+            C.AdaAssetId -> QueryByAssetId "" (Just "") Nothing
+            C.AssetId policyId assetName -> QueryByAssetId policyId (Just assetName) Nothing
+    -- Let's us show the actual error when initially getting the following test fail:
+    -- SQLite3 returned ErrorMisuse while attempting to perform prepare "SELECT slotNo, blockHeaderHash, txIndexInBlock, txId, policyId, assetName, quantity, redeemerHash, redeemerData FROM minting_policy_events WHERE slotNo <= :slotNo  AND policyId = :policyId AND assetName = :assetName ORDER BY slotNo ASC, txIndexInBlock ASC": bad parameter or other API misuse
+    H.annotateShow query
+    (sqlIndexerEvents :: [Core.Timed C.ChainPoint MintTokenBlockEvents]) <-
+      H.evalM $ H.evalExceptT $ Core.queryLatest query sqlIndexer
+    sqlIndexerEvents === []
+
+{- | Runs the 'MintTokenEvent' worker while providing 'AssetId's to track and events to index, and
+ return the resulting indexer.
+-}
+indexWithRunner
+  :: [C.AssetId]
+  -> [Core.Timed C.ChainPoint (Maybe MintTokenBlockEvents)]
+  -> PropertyT IO StandardMintTokenEventIndexer
+indexWithRunner trackedAssetIds events = do
+  (indexerVar, worker) <-
+    H.evalExceptT $
+      MintTokenEvent.mkMintTokenEventWorker
+        "MintTokenEventWorker"
+        (Core.CatchupConfig 4 2)
+        (MintTokenEventConfig trackedAssetIds)
+        id
+        ":memory:"
+
+  -- We create a coordinator to perform indexing through the worker
+  coordinator <- liftIO $ Core.mkCoordinator [worker]
+  let eventsWithDistance = fmap (fmap (Just . WithDistance 0)) $ reverse events
+  void $ H.evalExceptT $ Core.indexAllDescending eventsWithDistance coordinator
+  liftIO $ Concurrent.readMVar indexerVar
+
+-- | Query the events of the 'ListIndexer' and only keep the events that contain tracked AssetId.
+queryListIndexerEventsMatchingTargetAssetIds
+  :: [C.AssetId]
+  -> Core.ListIndexer MintTokenBlockEvents
+  -> IO [Core.Timed C.ChainPoint MintTokenBlockEvents]
+queryListIndexerEventsMatchingTargetAssetIds assetIds indexer = do
+  let query = Core.EventsMatchingQuery (filterByTargetAssetIds assetIds)
+  r <- runExceptT $ Core.queryLatest query indexer
+  either throwIO pure r
```

Comment:
We may also want to switch to `evalExceptT` and `PropertyT IO` here for consistency.